onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -692,13 +696,27 @@ def create_multihead_attention_node(
         else:
             mha_inputs.append("")
 
+        if (
+            past_k in graph_input_names
+            and past_v in graph_input_names
+            and present_k in graph_output_names
+            and present_v in graph_output_names
+        ):
+            kv_cache_name_match = True
+        else:
+            kv_cache_name_match = False
+
         # Add optional inputs for MHA
-        if past_k and past_v and past_k in graph_input_names and past_v in graph_input_names:
+        if past_k and past_v and kv_cache_name_match:
```

Comment:
This logic change 
```
        if past_k and past_v and past_k in graph_input_names and past_v in graph_input_names:
            mha_inputs.extend([key_padding_mask, add_qk, past_k, past_v])
   ```
   to 
   ```
     if past_k and past_v:
            mha_inputs.extend([key_padding_mask, add_qk, past_k, past_v])
   ```
   might impact existing model. Could you only change the logic for CT model, but not impact other model?

Reply:
Makes sense will update with the suggested logic. One other option is to write same function for the CT model but that will have lot of duplicate code